design/user_types.go

```diff
@@ -38,6 +38,71 @@ var UpdateWorkItemPayload = a.Type("UpdateWorkItemPayload", func() {
 	a.Required("type", "fields", "version")
 })
 
+// UpdateWorkItemJSONAPIPayload defines top level structure from jsonapi specs
+// visit : http://jsonapi.org/format/#document-top-level
+var updateWorkItemJSONAPIPayload = a.Type("UpdateWorkItemJSONAPIPayload", func() {
+	a.Attribute("data", workItemDataForUpdate)
+	a.Required("data")
+})
+
+// WorkItemDataForUpdate defines how an update payload will look like
+var workItemDataForUpdate = a.Type("WorkItemDataForUpdate", func() {
+	a.Attribute("type", d.String, func() {
+		a.Enum("workitems")
+	})
+	a.Attribute("id", d.String, "ID of the work item which is being updated", func() {
+		a.Example("42")
+	})
+	a.Attribute("attributes", a.HashOf(d.String, d.Any), func() {
+		a.Example(map[string]interface{}{"version": "1", "system.state": "new", "system.title": "Example story"})
+	})
+	a.Attribute("relationships", workItemRelationships)
+	// relationships must be required becasue we MUST have workItemType during PATCh
+	a.Required("type", "id", "attributes")
```

Comment:
Thought behind keeping it Required is - "Version" is taken from attributes and that is required field, so while `Update` we must have version.
As mentioned in the [description](https://github.com/almighty/almighty-core/pull/482#issue-188751436) we need "version" to update, and that is in attributes. So that was the idea behind keeping it required.